cloud-config/README.md

```diff
@@ -1,6 +1,135 @@
+    assembled templates.
+
+    We can save Assembled Templates with the --save-configure bin/deploy argument.  It will 
+    automacitallly determine which template to used based on input arguments.
+
+    $ bin/deploy --save-config-name 20200430
+
+    ### Output
+    Created assembeled template
+           ./cloud-config/assembled-templates/20200430-app-es-pg.yml
+    Deploy with
+           $ bin/deploy --use-prebuilt-config 20200430-app-es-pg
+    Diff with on the fly assembly.  Does not deploy
+           $ bin/deploy --use-prebuilt-config 20200430-app-es-pg --diff-configs
+    ###
+
+
+#### Directories:
+    template-parts              : Pieces of the templates
+    run-scripts                 : Install scripts runcmd_* template parts
+    configs                     : Configuration files used in run-scripts, like apache, java, es
+    assembled-templates         : Saved.  These still contains Run Varialbes
+
+#### Helper Script
+    create-ami.py               : Create amis from deployed --ami-build in AWS
+
+#### Run_Variables
+    * Run variables are in /etc/environment file on the instance.  
+    * They are used in the run-scripts to configure the system and application builds
+    * /etc/environment is loaded into login/ssh sessions so you can echo them on the instance.
+    * The file will contain dupicate entries when deploying from an AMI.  Last ones are used.
+    
+    View them locally with --dry-run along with other info.  Does not deploy
+    $ bin/deploy --dry-run
+
+    Add options like --test, --release-candidate, or --candidate to see the differences in run vars.
+    The ROLE should change along with other variables like ENCD_INDEX_PRIMARY, ENCD_INDEX_VIS,
+    ENCD_INDEX_REGION.  The env vars are prefixed with ENCD_ as to not conflict with other env vars.
+
+## Live Deployments
+    Here we'll deploy some instances using the --full-build args to avoid needing amis.  Building
+    from scratch is easier for cloud config development.
+
+#### QA/Development demo: app-es-pg-template.yml
+    $ bin/deploy --full-build
+
+    ### Output
+    Deploying app-es-pg
+    $ bin/deploy --full-build
+    create instance and wait for running state
+    ####
+
+#### Demo Cluster(es-wait): es-nodes-template.yml and app-pg-template
+    $ export CLUSTER_NAME='encd-dev'
+    $ bin/deploy --full-build --cluster-name "$CLUSTER_NAME" --es-wait
+   
+    ### Output
+    Deploying es-nodes
+    $ bin/deploy --full-build --cluster-name "$CLUSTER_NAME" --es-wait
+    Create instance and wait for running state
+    # Deploying Head ES Node(172.31.26.236): encd-dev-datamaster
+    ###
+
+    The IP address is the --es-ip used to deploy a frontend.
+    $ export ES_IP='172.31.26.236'
+
+    $ bin/deploy --full-build --cluster-name "$CLUSTER_NAME" --es-ip "$ES_IP"
+    
+    ### Output
+    Deploying app-pg
+    $ bin/deploy --full-build --cluster-name encd-dev --es-ip 172.31.26.236
```

Comment:
again not sure why this is repeated (unless it is the echo from running the command)